app/components/meeting/edit-meeting-modal.js

```diff
@@ -47,40 +68,175 @@ export default class MeetingEditMeetingComponent extends Component {
 
   set meetingNumber(meetingNumber) {
     this._meetingNumber = meetingNumber;
-    this._numberRepresentation = null;
+    if (meetingNumber) {
+      this._numberRepresentation = null;
+    }
   }
 
+  get savingIsDisabled() {
+    return (
+        (this.isAnnexMeeting && !this.selectedMainMeeting) ||
+        !this.meetingNumber ||
+        !this.numberRepresentation ||
+        this.initializeKind.isRunning ||
+        this.initializeMainMeeting.isRunning ||
+        this.saveMeeting.isRunning
+      );
+  }
+
+  @task
+    *initializeMainMeeting() {
+      this.selectedMainMeeting = yield this.args.meeting.mainMeeting;
+    }
+
   @task
   *initializeKind() {
     this.kind = yield this.args.meeting.kind;
+    this.kind ??= yield this.store.findRecordByUri('concept', CONSTANTS.MEETING_KINDS.MINISTERRAAD);
+    const broader = yield this.kind?.broader;
+    this.isAnnexMeeting = broader?.uri === CONSTANTS.MEETING_KINDS.ANNEX;
+  }
+
+  @dropTask
+  *initializeMeetingNumber() {
+    if (this.args.meeting.number) {
+      this.meetingNumber = this.args.meeting.number;
+    } else {
+      const meeting = yield this.store.queryOne('meeting', {
+        filter: {
+          ':gte:planned-start': new Date(this.currentYear, 0, 1).toISOString(),
+          ':lt:planned-start': new Date(this.currentYear + 1, 0, 1).toISOString(),
+        },
+        sort: '-number',
+      });
+
+      const id = meeting?.number ?? 0;
+      this.meetingNumber = id + 1;
+    }
   }
 
-  @task({ drop: true })
-  *updateMeeting() {
+  @dropTask
+  *saveMeeting() {
+    const isNewMeeting = this.args.meeting.isNew;
     const now = new Date();
 
     this.args.meeting.extraInfo = this.extraInfo;
     this.args.meeting.plannedStart = this.startDate || now;
     this.args.meeting.kind = this.kind;
     this.args.meeting.number = this.meetingNumber;
     this.args.meeting.numberRepresentation = this.numberRepresentation;
+    this.args.meeting.mainMeeting = this.selectedMainMeeting;
+    this.args.meeting.isFinal = false;
```

Comment:
A bit weird to overwrite this flag while editing the meeting. Was it like this before?

Reply:
was meant to use `this.isFinal` here, thanks for catching that!